### PR TITLE
Cleanup for TreeBrowserBundle 2.0

### PR DESCRIPTION
**Pull request by WouterJ**
A cleanup of the bundle for the treebrowserbundle cleanup. Atm, the bundle is broken. This fixes it.

It contains some big BC breaks (removing tree for instance). Should I note this in the CHANGELOG file? (which doesn't contain any 2.0 changes atm)

Other things to be aware of:

* This now means that valid_children is no longer used. It's not very hard to implement it for the "create" action, but for displaying it requires new features in the resource bundle (/cc @dantleech am I correct or does it already have such feature?)
* Some options are now commented, I wanted to leave them in so we see what's still missing.
* Selected node is now no longer needed, as the treebrowser supports binding to the input field (automatically selecting based on the input field)

### CODE REVIEW

File: Controller/TreeController.php [hunk 1]
```diff
@@ -22,9 +21,9 @@
 class TreeController extends Controller
```
Comment (dbu) [inline Controller/TreeController.php]: do we still need such a controller in this bundle or will this all move to the tree bundle?

File: Tree/PhpcrOdmTree.php [hunk 2]
```diff
@@ -1,470 +0,0 @@
...
- * A tree implementation to work with Doctrine PHPCR-ODM
- *
- * Your documents need to map all children with an Children mapping for the
- * tree to see its children. Not having the Children annotation is a
- * possibility to not show children you do not want to show.
- *
- * @author David Buchmann <david@liip.ch>
- * @author Uwe Jäger <uwej711@googlemail.com>
- */
-class PhpcrOdmTree implements TreeInterface
```
Comment (dbu) [inline Tree/PhpcrOdmTree.php]: according to travis there is a test for this class that now produces a fatal error when running the tests...

**Comment by dbu** [comment]
cool! i am happy to merge as soon as the tests are cleaned up. we should create issues for the currently abandoned functionality like valid_children however, so that we do not forget this.

and yes, lets add things to CHANGELOG.md and if possible also UPGRADE.md. doing this later will become much harder when we don't remember that well what we did.

File: Controller/TreeController.php (hunk 1 above)
Comment (WouterJ) [inline Controller/TreeController.php]: The controller is for the tree block, which is still used.

**Comment by WouterJ** [comment]
Fixed the tests and updated the changelog.

**Comment by dbu** [comment]
thanks a lot! as soon as it becomes clear, we should start writing a migration guide as well. but maybe wait until its more clear how the new things will work exactly.

**Comment by WouterJ** [comment]
What do you mean with "a migration guide"?

**Comment by dbu** [comment]
instructions how to upgrade a project from using admin 1.x to use admin 2.0. "upgrade guide" would be a better name probably.

**Pull request merged**